doc/tensor.md

```diff
@@ -1327,36 +1331,64 @@ x:indexFill(2,torch.LongTensor{4,2},-10)
 ```
 
 <a name="torch.Tensor.gather"/>
-### [Tensor] gather(src, dim, index) ###
+### [Tensor] torch.gather(src, dim, index) ###
 
-Reads values from `src` at the specified indices into the output tensor, selecting a fixed number of values in each row of `src` along the specified dimension.
-The values of `index` must be between 1 and `src->size(dim)` and all values in a row along the specified dimension must be unique.
-All tensors have the same dimensionality and both `index` and the output tensor are of the same size at the specified dimension.
+Creates a new `Tensor` from the `src` tensor by gathering a number of values from each "row" of `src`, where the rows are
+along the dimension `dim`. The values in a `LongTensor`, passed as `index`, specify which values to take from each row.
+Specifically, the resulting `Tensor`, which will have the same size as the `index` tensor, is given by
+
+```lua
+-- dim = 1
+result[i][j][k]... = src[index[i][j][k]...][j][k]...
+
+-- dim = 2
+result[i][j][k]... = src[i][index[i][j][k]...][k]...
+
+-- etc.
+```
+
+The same number of values are selected from each row or column, and the same value cannot be selected from a row or column
+more than once. The values in the `index` tensor must be smaller than the length of the row or column, that is between
+1 and `src:size(dim)`. It can be somewhat confusing to ensure that the `index` tensor has the correct shape. Viewed
```

Comment:
In the first paragraph you only talk about rows, but here you say "row or column". Did you mean to do that or did you just forget to change that from your earlier version?
Otherwise it looks good, thanks!